### ISSUE TITLE
IndexOutOfBoundExeption when notifyDataSetChanged() is called

### ISSUE THREAD

**Issue by alenjularic** [nitrico/LastAdapter#28]
Stack trace:
```
07-10 15:11:53.936 E/AndroidRuntime: FATAL EXCEPTION: main
                                     Process: com.nejctomsic.registerzdravil.debug, PID: 18830
                                     java.lang.IndexOutOfBoundsException: Index: 3, Size: 1
                                         at java.util.ArrayList.get(ArrayList.java:411)
                                         at com.github.nitrico.lastadapter.LastAdapter.getType(LastAdapter.kt:178)
                                         at com.github.nitrico.lastadapter.LastAdapter.onViewRecycled(LastAdapter.kt:132)
                                         at com.github.nitrico.lastadapter.LastAdapter.onViewRecycled(LastAdapter.kt:27)
                                         at android.support.v7.widget.RecyclerView$Recycler.dispatchViewRecycled(RecyclerView.java:6064)
```
The issue was not present until version 2.2.2. It was added in [this commit](https://github.com/nitrico/LastAdapter/commit/22055799a6c36f71a741c5a243fe92be3f3d5472#diff-96125d8176af46d9606b163013568db2).  The problem is when notifyDataSetChanged() is called on the adapter, with a new backing list that has fewer elements than the old one. To fix this, the [line that compares the position](https://github.com/nitrico/LastAdapter/blob/master/lastadapter/src/main/java/com/github/nitrico/lastadapter/LastAdapter.kt#L131) should be reverted to [this one](https://github.com/nitrico/LastAdapter/blob/415eb276a877e5f52917435b5bf046ebdaec461d/lastadapter/src/main/java/com/github/nitrico/lastadapter/LastAdapter.kt#L128).

**Comment by nitrico** [comment]
Hi @alenjularic 

I've never seen this problem and no one else reported it. Can you please make sure you are using it correctly? I'll check it as well. Can you please let me know if you're using an ObservableList or just a regular List?

Thanks.

**Comment by dandanx** [comment]
Same problem here, it happens only if I filter the data (less data in list) and call notifyDataSetChanged()

`
Caused by: java.lang.IndexOutOfBoundsException: Index: 2, Size: 1
   at java.util.ArrayList.get(ArrayList.java:411)
   at com.github.nitrico.lastadapter.LastAdapter.getType(LastAdapter.kt:178)
   at com.github.nitrico.lastadapter.LastAdapter.onViewRecycled(LastAdapter.kt:132)
   at com.github.nitrico.lastadapter.LastAdapter.onViewRecycled(LastAdapter.kt:27)
   at android.support.v7.widget.RecyclerView$Recycler.dispatchViewRecycled(RecyclerView.java:6182)
`

**Comment by VJYMAK** [comment]
Same problem here. When the list is filtered with the less data than the previous one. 
```
java.lang.IndexOutOfBoundsException: Inconsistency detected. Invalid item position 10(offset:10).state:50 android.support.v7.widget.RecyclerView{a6d485 VFED..... ........ 0,0-1080,1548 #7f0900c4 app:id/rcv_news_latest}, adapter:com.github.nitrico.lastadapter.LastAdapter@fa732da, layout:android.support.v7.widget.LinearLayoutManager@a4b430b, context:com.fintech.assetvantage.main.views.MainActivity@1101dba
                                                                            at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5628)
                                                                            at android.support.v7.widget.GapWorker.prefetchPositionWithDeadline(GapWorker.java:285)
                                                                            at android.support.v7.widget.GapWorker.flushTaskWithDeadline(GapWorker.java:342)
                                                                            at android.support.v7.widget.GapWorker.flushTasksWithDeadline(GapWorker.java:358)
                                                                            at android.support.v7.widget.GapWorker.prefetch(GapWorker.java:365)
                                                                            at android.support.v7.widget.GapWorker.run(GapWorker.java:396)
                                                                            at android.os.Handler.handleCallback(Handler.java:751)
                                                                            at android.os.Handler.dispatchMessage(Handler.java:95)
```

**Comment by dogs-never-lie** [comment]
Same problem. Removing item from list and causes IndexOfBound.

**Comment by VJYMAK** [comment]
I still don't understand why this exception occurred but when i change my list to observable list my problem is solve.
@nitrico Please check this issue it's really a major issue.

**Comment by nitrico** [comment]
I'm sorry it took so long... Can you guys please check if the problem is solved in version 2.3.0?

**Comment by alenjularic** [comment]
No problem @nitrico. Yes it's fixed for me in 2.3.0. Thank you so much!
